spec/pluralrules.html

```diff
@@ -0,0 +1,360 @@
+             <td>Number</td>
+             <td>Number of visible fraction digits in [[Number]], <i>with</i> trailing zeros.</td>
+           </tr>
+           <tr>
+             <td>[[NumberOfFractionDigitsWithoutTrailing]]</td>
+             <td>Number</td>
+             <td>Number of visible fraction digits in [[Number]], <i>without</i> trailing zeros.</td>
+           </tr>
+           <tr>
+             <td>[[FractionDigits]]</td>
+             <td>Number</td>
+             <td>Number of visible fractional digits in [[Number]], <i>with</i> trailing zeros.</td>
+           </tr>
+           <tr>
+             <td>[[FractionDigitsWithoutTrailing]]</td>
+             <td>Number</td>
+             <td>Number of visible fractional digits in [[Number]], <i>without</i> trailing zeros. </td>
+           </tr>
+         </table>
+       </emu-table>
+
+    </emu-clause>
+
+    <emu-clause id="sec-pluralruleselection" aoid="PluralRuleSelection">
+
+      <h1>PluralRuleSelection ( _locale_, _type_, _n_, _operands_ )</h1>
+
+      <p>
+        When the PluralRuleSelection abstract operation is called with four arguments, it performs implementation dependent algorithm to map _n_ to the appropriate plural representation of the Operand Record _operands_ by selecting the rules denoted by _type_ for the corresponding _locale_, or the String value `"other"`.
+      </p>
+
+    </emu-clause>
+
+    <emu-clause id="sec-resolveplural" aoid="ResolvePlural">
+      <h1>ResolvePlural ( _pluralRules_, _n_ )</h1>
+      <p>
+        When the ResolvePlural abstract operation is called with arguments _pluralRules_ (which must be an object initialized as a PluralRules) and _n_ (which must be a Number value), it returns a String value representing the plural form of _n_ according to the effective locale and the options of _pluralRules_. The following steps are taken:
+      </p>
+
+      <emu-alg>
+        1. Assert: Type(_pluralRules_) is Object.
+        1. Assert: If _pluralRules_ does not have an [[InitializedPluralRules]] internal slot, throw a *TypeError* exception.
+        1. Assert: Type(_n_) is Number.
+        1. If isFinite(_n_) is *false*, then
+          1. Return `"other"`.
+        1. Let _locale_ be _pluralRules_.[[Locale]].
+        1. Let _type_ be _pluralRules_.[[Type]].
+        1. Let _s_ be ? FormatNumberToString(_pluralRules_, _n_).
+        1. Let _operands_ be ? _GetOperands_(_s_).
+        1. Return ? PluralRuleSelection(_locale_, _type_, _n_, _operands_).
+      </emu-alg>
+
+    </emu-clause>
+
+  </emu-clause>
+
+  <emu-clause id="sec-intl-pluralrules-constructor">
+    <h1>The Intl.PluralRules Constructor</h1>
+
+    <p>
+      The PluralRules constructor is the <dfn>%PluralRules%</dfn> intrinsic object and a standard built-in property of the Intl object. Behaviour common to all service constructor properties of the Intl object is specified in <emu-xref href="#sec-internal-slots"></emu-xref>.
```

Comment:
- https://tc39.github.io/ecma402/#sec-internal-slots also needs to be updated to list `Intl.PluralRules`.
- Is the DefaultLocale restriction also okay for PluralRules? That means a certain locale can only be the default locale if this locale is also supported by the PluralRules constructor.

Reply:
yes, DefaultLocale should also apply to PluralRules IMO. I will ask tomorrow for clarification from the champions.

Reply:
AFAICT ICU, for one, supports PluralRules for all locales where CLDR has data, and doesn't actually let you configure which locales you want to ship PluralRules data for (maybe because it's so little data). So I don't think the DefaultLocale restriction should be a problem for ICU-based implementations.